indico/htdocs/js/indico/modules/contributions/common.js

```diff
@@ -28,11 +28,82 @@
         });
     }
 
+    function formatState(visible, total) {
+        return format($T.gettext('{0} / {1}'), [
+            format('<strong>{0}</strong>', [visible.length]),
+            total.length
+        ])
+    }
+
+    function setState(visible, total) {
+        var $state = $('#filtering-state'),
+            title = format('{0} out of {1} contributions displayed', [visible.length, total.length]);
+        $state.html(formatState(visible, total));
+
+        // oldtitle needs to be updated too, because of qTip
+        $state.attr({
+            oldtitle: title,
+            title, title
+        });
+    }
+
+    function applyFilters() {
+        var contributions = $('#contribution-list tbody tr'),
+            term = $('#search-input').val().trim(),
+            visibleEntries,
+            $state = $('#filtering-state'),
+            $filterPlaceholder = $('#filter-placeholder');
+
+        $filterPlaceholder.hide();
+        $state.removeClass('active');
+
+        if (term == '') {
+            contributions.show();
+            setState(contributions, contributions);
+            return;
+        }
+
+        // quick search of contribution by ID
+        if (term.match(/^#\d+$/)) {
+            visibleEntries = contributions.find('.id-column').textContains(term).closest('tr');
```

Comment:
I think this will match `#123` when you enter `#1` (not sure if that's a good idea or not)